### PR TITLE
Calico updates to v3.0 and add auto-scalers

### PR DESCRIPTION
**Pull request by tmjd**

### CODE REVIEW

File: misc/aws-k8s-cni-calico.yaml [hunk 1]
```diff
@@ -79,12 +83,6 @@ spec:
     spec:
       hostNetwork: true
       serviceAccountName: calico-node
```
Comment (tmjd) [inline misc/aws-k8s-cni-calico.yaml:85]: Does calico-node still need access to the datastore when using typha?
Right now calico-node and typha use the same serviceAccount, any reason to give them separate ones.

File: misc/aws-k8s-cni-calico.yaml [hunk 2]
```diff
@@ -151,48 +157,57 @@ spec:
...
           hostPath:
             path: /lib/modules
         - name: var-run-calico
           hostPath:
             path: /var/run/calico
+      tolerations:
+        # Make sure calico/node gets scheduled on all nodes.
+        - effect: NoSchedule
+          operator: Exists
+        - effect: NoExecute
```
Comment (tmjd) [inline misc/aws-k8s-cni-calico.yaml:171]: Should we pick this change up from what has been done to the Calico manifests? Seems reasonable to me but I wanted to bring it up specifically.

**Review by tmjd — COMMENTED** [review]
I basically picked up the changes that have happened to the kdd/policy-only manifests since v2.6.3 to v3.0 and then added the auto-scalers.

File: misc/aws-k8s-cni-calico.yaml (hunk 1 above)
Comment (caseydavenport) [reply misc/aws-k8s-cni-calico.yaml:85]: Yeah, they maybe could have different ones, but calico/node still needs data store access.

File: misc/aws-k8s-cni-calico.yaml (hunk 2 above)
Comment (caseydavenport) [reply misc/aws-k8s-cni-calico.yaml:171]: Yeah I think so

**Review by caseydavenport — APPROVED** [review]

**Comment by tmjd** [comment]
Closing this since it was against my own master to get agreement before submitting for 'real'.

**Pull request closed without merging**